### DISCUSSION TITLE
Deleting child records with a composite key via Hook

### DISCUSSION THREAD

**Discussion by mborgues** [TriPSs/nestjs-query#328] [General]
Hello,

How do I create a "BeforeDeleteOneHook" hook where I will delete the child records? My legacy database does not have the cascade option configured and therefore it will be necessary to delete the children before deleting the parent record. I have noticed that the DeleteOneInputType is not generic and therefore I cannot access my child ID, which is an object composed of the key fields. Is there an example to perform this process, or any resource other than using the Hook?

Thank you.

**Comment by TriPSs** [comment]
I'm having the some in one of our projects, what we do there is overwrite the query service like this:
```ts
@Injectable()
export class MyQueryService extends TypeOrmQueryService<MyEntity> {

  public async deleteOne(id: number, opts?: DeleteOneOptions<MyEntity>): Promise<MyEntity> {
    const record = await super.deleteOne(id, opts)

    await this.otherRepoOne.delete({ parentId: id })
    await this.otherRepoTwo.delete({ parentId: id })
    // etc

    return record
  }

}
```

**Reply by mborgues** [reply]
Could you explain this process to me better? Do I need to create a resolver? Or how do I use this queryService? I'm a bit lost on how to use this feature.

Thanks.

**Reply by TriPSs** [reply]
You can use this service inside the module by setting `ServiceClass: MyQueryService`, see [here](https://tripss.github.io/nestjs-query/docs/persistence/typeorm/custom-service/) for more info.
